src/main/java/com/avispl/symphony/dal/infrastructure/management/sembient/sembient/SembientAggregatorCommunicator.java

```diff
@@ -0,0 +1,1813 @@
+							throw new IllegalStateException("Too many requests sent to the device, please try to create one more time with value: " + newTag);
+						}
+						break;
+					case SembientAggregatorConstant.DELETE_SELECTED_TAG:
+						String valueToBeDelete = aggregatedDeviceTagMap.get(deviceId);
+						if (StringUtils.isNullOrEmpty(valueToBeDelete)) {
+							throw new IllegalArgumentException("Tag dropdowns value cannot be empty.");
+						}
+						String deleteRequest =
+								SembientAggregatorConstant.COMMAND_SPACE_TAGS + loginResponse.getCustomerId() + SembientAggregatorConstant.SLASH + buildingID + SembientAggregatorConstant.SLASH + floorName
+										+ SembientAggregatorConstant.PARAM_REGION_NAME + deviceName + SembientAggregatorConstant.PARAM_REGION_TAGS + valueToBeDelete;
+						try {
+							this.doDelete(deleteRequest);
+							// Get old tags
+							List<String> options = new ArrayList<>();
+							for (AdvancedControllableProperty control : controlFromCached) {
+								if (SembientAggregatorConstant.PROPERTY_TAG.equals(control.getName())) {
+									options = new LinkedList<>(Arrays.asList(((DropDown) control.getType()).getOptions()));
+									break;
+								}
+							}
+							// Remove deleted tags
+							options.remove(valueToBeDelete);
+							if (options.size() == 0) {
+								controlFromCached.removeIf(advancedControllableProperty -> advancedControllableProperty.getName().equals(SembientAggregatorConstant.PROPERTY_DELETE));
+								statFromCached.remove(SembientAggregatorConstant.PROPERTY_DELETE);
+								controlFromCached.removeIf(advancedControllableProperty -> advancedControllableProperty.getName().equals(SembientAggregatorConstant.PROPERTY_TAG));
+								statFromCached.remove(SembientAggregatorConstant.PROPERTY_TAG);
+								break;
+							}
+							controlFromCached.removeIf(advancedControllableProperty -> advancedControllableProperty.getName().equals(SembientAggregatorConstant.PROPERTY_TAG));
+							statFromCached.remove(SembientAggregatorConstant.PROPERTY_TAG);
+							aggregatedDeviceTagMap.put(deviceId, options.get(0));
+							controlFromCached.add(createDropdown(statFromCached, SembientAggregatorConstant.PROPERTY_TAG, options, options.get(0)));
+						} catch (CommandFailureException e) {
+							logger.error("Fail to delete with status code: " + e.getStatusCode() + ", value: " + valueToBeDelete, e);
+							if (e.getStatusCode() == 429) {
+								throw new IllegalStateException("Too many request, please try to delete with value: " + valueToBeDelete + " later.");
+							} else {
+								throw new IllegalStateException("Fail to delete region tag with value: " + valueToBeDelete);
+							}
+						} catch (Exception e) {
+							logger.error("Exception occur when deleting region tag with value: " + valueToBeDelete, e);
+							throw new IllegalStateException("Fail to delete region tag with value: " + valueToBeDelete);
```

Comment:
I rethrown it as IllegalStateException here is because I want the error message much clearer to the user, do you think it makes sense? Because if we throw original one here, it might hard to read.